### PR TITLE
Handwritten aeson instances

### LINKED ISSUE #472: API Types should have handwritten aeson instances
- [x] Manually define aeson `FromJSON` and `ToJSON` instances for types that are used in the API.
- [x] Add this as a note to the Style guide

### PR DESCRIPTION
**Pull request by sajidanower23**
Closes #472 
- Stack v2.1.1 compatibility for or_scs, mirza-common-haskell and entity-data-api.
- Updates GS1Combinators snapshot
- Introduces handwritten instances for all types that are used in the API
- Adds a note to the Style Guide

### CODE REVIEW

File: projects/or_scs/src/Mirza/OrgRegistry/Types.hs [hunk 1]
```diff
@@ -171,8 +171,8 @@ instance FromJSON PartialNewOrg where
     <*> o .: "url"
 
 instance ToJSON PartialNewOrg where
-  toJSON (PartialNewOrg orgName url) = object
-    [ "name" .= orgName
+  toJSON (PartialNewOrg orgname url) = object
```
Comment (a-stacey) [inline projects/or_scs/src/Mirza/OrgRegistry/Types.hs]: Why the change of case here? Feels more correct before?

File: projects/or_scs/src/Mirza/OrgRegistry/Types.hs [hunk 2]
```diff
@@ -200,9 +200,9 @@ instance FromJSON OrgResponse where
     <*> o .: "url"
 
 instance ToJSON OrgResponse where
-  toJSON (OrgResponse (EPC.GS1CompanyPrefix pfix) orgName url) = object
+  toJSON (OrgResponse pfix orgname url) = object
```
Comment (a-stacey) [inline projects/or_scs/src/Mirza/OrgRegistry/Types.hs]: Case change on orgName?

File: projects/or_scs/src/Mirza/OrgRegistry/Types.hs [hunk 3]
```diff
@@ -291,29 +291,58 @@ data LocationResponse = LocationResponse
...
+    "expired" -> pure Expired
+    _ -> fail "Invalid value for key_state. Valid values are in_effect, revoked and expired."
+instance ToJSON KeyState where
+  toJSON InEffect = String "in_effect"
+  toJSON Revoked  = String "revoked"
+  toJSON Expired  = String "expired"
 instance ToSchema KeyState
 instance ToParamSchema KeyState
 
-
```
Comment (a-stacey) [inline projects/or_scs/src/Mirza/OrgRegistry/Types.hs]: Pedantic, but I think its good to have 2 spaces here. I've been trying to add 2 spaces between functions/logical groups and use single spaces within, just to assist with grouping readability.

File: projects/or_scs/src/Mirza/OrgRegistry/Types.hs (hunk 1 above)
Comment (sajidanower23) [reply projects/or_scs/src/Mirza/OrgRegistry/Types.hs]: Because `orgName` is an accessor function in `OrgResponse`.
```haskell
data OrgResponse = OrgResponse
  { orgGS1CompanyPrefix :: EPC.GS1CompanyPrefix
  , orgName             :: Text
  , orgUrl              :: Network.URI.URI
  }
```

File: projects/or_scs/src/Mirza/OrgRegistry/Types.hs (hunk 3 above)
Comment (sajidanower23) [reply projects/or_scs/src/Mirza/OrgRegistry/Types.hs]: Oh, apologies. I wasn't aware of that. I will fix that.

File: projects/or_scs/src/Mirza/OrgRegistry/Types.hs [hunk 4]
```diff
@@ -291,29 +291,58 @@ data LocationResponse = LocationResponse
...
+instance ToJSON OrgAndLocationResponse where
+  toJSON (OrgAndLocationResponse org loc) = object
+    [ "org" .= org
+    , "loc" .= loc
+    ]
+instance FromJSON OrgAndLocationResponse where
+  parseJSON = withObject "OrgAndLocationResponse" $ \o -> OrgAndLocationResponse
+    <$> o .: "org"
+    <*> o .: "location"
```
Comment (a-stacey) [inline projects/or_scs/src/Mirza/OrgRegistry/Types.hs]: These don't appear to be complimentary: "loc" vs "location". For reference in the public interface I strongly believe that why shouldn't be abbreviating as I think this aids with readability (even if there are other considerations like bandwitdh etc). Another thing to consider is to used "named constants" (or similar technique) to make sure that the JSON key strings are the same in toJSON and fromJSON, to prevent this type of issue happening. This comment applies everywhere. e.g.
```
orgAndLocationResponseOrgKey = "org"
orgAndLocationResponseLocKey = "loc"

instance ToJSON OrgAndLocationResponse where
  toJSON (OrgAndLocationResponse org loc) = object
    [ orgAndLocationResponseOrgKey .= org
    , orgAndLocationResponseLocKey .= loc
    ]
instance FromJSON OrgAndLocationResponse where
  parseJSON = withObject "OrgAndLocationResponse" $ \o -> OrgAndLocationResponse
    <$> o .: orgAndLocationResponseOrgKey
    <*> o .: orgAndLocationResponseLocKey
```
